src/charm.py

```diff
@@ -139,7 +139,15 @@ def __init__(self, *args):
             self,
             relation_name="self-metrics-endpoint",
             jobs=self.self_scraping_job,
-            external_url=self._external_url,
+            refresh_event=[
+                self.on.update_status,
+                self.ingress.on.ready,
+                self.ingress.on.revoked,
+                self.on["ingress"].relation_changed,
+                self.on["ingress"].relation_departed,
```

Comment:
Do we need ingress refresh events if we're always taking the fqdn?

Reply:
No, we are not taking always the fdqn, if the charm is related to traefik, we use the external_url provided.

```python
    @property
    def _external_url(self) -> str:
        """Return the externally-reachable (public) address of the alertmanager api server."""
        return self.model.config.get("web_external_url") or self.ingress.url or self._internal_url
```